src/components/TextInput/BaseTextInput/index.native.tsx

```diff
@@ -21,36 +35,70 @@ import useTheme from '@styles/themes/useTheme';
 import useThemeStyles from '@styles/useThemeStyles';
 import variables from '@styles/variables';
 import CONST from '@src/CONST';
-import * as baseTextInputPropTypes from './baseTextInputPropTypes';
-
-function BaseTextInput(props) {
+import BaseTextInputProps from './types';
+
+function BaseTextInput(
+    {
+        label = '',
+        // name = '',
+        value = undefined,
+        defaultValue = undefined,
+        placeholder = '',
+        errorText = '',
+        icon = null,
+        textInputContainerStyles = [],
+        containerStyles = [],
+        inputStyle = [],
+        forceActiveLabel = false,
+        autoFocus = false,
+        disableKeyboard = false,
+        autoGrow = false,
+        autoGrowHeight = false,
+        hideFocusedState = false,
+        maxLength = undefined,
+        hint = '',
+        // shouldSaveDraft = false,
+        onInputChange = () => {},
+        shouldDelayFocus = false,
+        submitOnEnter = false,
+        multiline = false,
+        // shouldUseDefaultValue = false,
+        shouldInterceptSwipe = false,
+        autoCorrect = true,
+        prefixCharacter,
+        inputID,
+        ...inputProps
+    }: BaseTextInputProps,
+    ref: ForwardedRef<HTMLFormElement | Component<AnimatedProps<TextInputProps>, unknown, unknown>>,
+) {
     const theme = useTheme();
     const styles = useThemeStyles();
-    const initialValue = props.value || props.defaultValue || '';
-    const initialActiveLabel = props.forceActiveLabel || initialValue.length > 0 || Boolean(props.prefixCharacter);
-    const isMultiline = props.multiline || props.autoGrowHeight;
+    const {hasError = false} = inputProps;
+    const initialValue = value ?? defaultValue ?? '';
+    const initialActiveLabel = !!forceActiveLabel || initialValue.length > 0 || Boolean(prefixCharacter);
```

Comment:
Since you already use `!!` in this line
```suggestion
    const initialActiveLabel = !!forceActiveLabel || initialValue.length > 0 || !!prefixCharacter;
```